Sources/Tracking/Background Queue/QueueStorage.swift

```diff
@@ -104,6 +105,19 @@ public class FileManagerQueueStorage: QueueStorage {
 
         return task
     }
+
+    public func delete(storageId: String) -> Bool {
```

Comment:
Mind giving an example use case to demonstrate your concern?

Reply:
Basically it's an issue with in-memory state & time slicing in the OS.

`get` / `getInventory` case (probably not a problem, since dequeuing is serial)

```
given `item1` in queue
1. delete initiated
2. separate thread calls one of `get` / `getInventory` -> inventory contains `item1`
3. delete commits -> inventory empty
4. separate thread uses result of `get` in some way, sees `item1` which should be deleted
```

`create` (one option, creates trash)
```
given `item1` in queue
1. delete initiated, calls `getInventory` -> inventory contains `item1` 
2. create initiated, calls `getInventory` -> inventory contains `item1`
3. delete completes -> inventory is empty
4. create completes -> inventory contains `item1` and `item2`
```


`create` (worse option, data loss)

```
given `item1` in queue
1. delete initiated, calls `getInventory` -> inventory contains `item1` 
2. create called -> inventory contains `item1` and `item2`
3. delete completes -> inventory is empty
```

for each of the above, a numeric line is what thread the OS is actively working on (assuming a single core & timesharing for simplicity, same problem exists regardless of number of cores but it's easier to describe this way). Cursory research only shows that iOS is based on darwin which is unix-y, so I think it's prudent to assume it'll behave this way.

Reply:
Mutex added in [commit](https://github.com/customerio/customerio-ios/pull/78/commits/018b666f5cd078c895a7e3bb9538b5ae75aa1882)